torch/jit/__init__.py

```diff
@@ -449,32 +452,26 @@ def trace(*args, **kwargs):
         ... def f(x):
         ...     return x * 2
     """
-    def wrapper(func):
-        if not _enabled:
-            return func
-        executor_options = {'optimize': True}
-        for name in executor_options:
-            executor_options[name] = kwargs.pop(name, executor_options[name])
-
-        check_inputs = kwargs.pop('check_inputs', None)
-        disable_checks = kwargs.pop('disable_checks', False)
-        check_tolerance = kwargs.pop('check_tolerance', 1e-5)
-
-        if len(kwargs) != 0:
-            raise TypeError("got unexpected keyword arguments: {}".format(", ".join(kwargs.keys())))
-
-        module = TopLevelTracedModule(func, **executor_options)
-        module._create_method_from_trace('forward', func, tuple(args))
-
-        # Check the trace against new traces created from user-specified inputs
+    if not _enabled:
+        return func
+    executor_options = {'optimize': bool(optimize)}
+    # Special case for common case of passing a single Variable
+    if isinstance(example_inputs, torch.Tensor):
+        example_inputs = (example_inputs,)
+    # done primarily so that weird iterables fail here and not pybind11 code
+    if not isinstance(example_inputs, tuple):
+        example_inputs = tuple(e for e in example_inputs)
+    module = TopLevelTracedModule(func, **executor_options)
+    module._create_method_from_trace('forward', func, example_inputs)
+
+    # Check the trace against new traces created from user-specified inputs
+    if check_trace:
```

Comment:
Why are we making the checks optional? We definitely should have some level of analysis enabled by default. It might not be numerical checks, but still.